Sources/StreamChatUI/CommonViews/Attachments/AttachmentViews/VoiceRecordingAttachmentComposerPreview.swift

```diff
@@ -0,0 +1,180 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// A view that displays the VoiceRecording attachment.
+open class VoiceRecordingAttachmentComposerPreview: _View, AppearanceProvider, ComponentsProvider, AudioPlayingDelegate {
+    public struct Content {
+        /// The title of the attachment.
+        public var title: String
+
+        /// The size of the attachment.
+        public var size: Int64
+
+        /// The recording's duration
+        public var duration: TimeInterval
+
+        /// The local or remote URL to the file
+        public var audioAssetURL: URL
+
+        public init(
+            title: String,
+            size: Int64,
+            duration: TimeInterval,
+            audioAssetURL: URL
+        ) {
+            self.title = title
+            self.size = size
+            self.duration = duration
+            self.audioAssetURL = audioAssetURL
+        }
+    }
+
+    public var content: Content? {
+        didSet {
+            updateContentIfNeeded()
+        }
+    }
+
+    /// The height the previewView should have
+    open var height: CGFloat = 54
+
+    /// The audioPlayer that will be used for the recording's playback.
+    ///
+    /// - Note: Upon set, the view will subscribe to receive playback events from the audioPlayer.
+    open var audioPlayer: AudioPlaying? {
+        didSet { audioPlayer?.subscribe(self) }
+    }
+
+    /// It provides the index of the recording in the containing message's attachment array.
+    open var indexProvider: (() -> Int)?
+
+    open private(set) lazy var container: UIStackView = .init()
+        .withoutAutoresizingMaskConstraints
+
+    open private(set) lazy var playPauseButton: PlayPauseButton = .init()
+        .withoutAutoresizingMaskConstraints
+        .withAccessibilityIdentifier(identifier: "actionButton")
+
+    open private(set) lazy var fileNameLabel = UILabel()
+        .withoutAutoresizingMaskConstraints
+        .withBidirectionalLanguagesSupport
+        .withAdjustingFontForContentSizeCategory
+        .withAccessibilityIdentifier(identifier: "fileNameLabel")
+
+    open private(set) lazy var durationLabel = UILabel()
+        .withoutAutoresizingMaskConstraints
+        .withBidirectionalLanguagesSupport
+        .withAdjustingFontForContentSizeCategory
+        .withAccessibilityIdentifier(identifier: "durationLabel")
+
+    open private(set) lazy var fileNameAndDurationStack: ContainerStackView = {
+        let stack = ContainerStackView(arrangedSubviews: [fileNameLabel, durationLabel])
+        stack.axis = .vertical
+        stack.alignment = .leading
+        stack.spacing = 3
+        return stack
+            .withoutAutoresizingMaskConstraints
+            .withAccessibilityIdentifier(identifier: "fileNameAndDurationStack")
+    }()
```

Comment:
As mentioned in a previous PR, usually it is best if we provide a more generic name for the stack 👍